File: exercise/ts/day13/tests/FakeCatalog.ts
import {SantamarketCatalog} from "../src/santamarket.model/SantamarketCatalog";
import {Product} from "../src/santamarket.model/Product";

export class FakeCatalog implements SantamarketCatalog {
    private prices: Map<string, number> = new Map();

    addProduct(product: Product, price: number): void {
        this.prices.set(product.name, price);
    }

    getUnitPrice(product: Product): number {
        return this.prices.get(product.name) || 0;
    }
}

File: exercise/python/day13/tests/fake_catalog.py
from model.santamarket_catalog import SantamarketCatalog


class FakeCatalog(SantamarketCatalog):
    def __init__(self):
        self.products = {}
        self.prices = {}

    def add_product(self, product, price: float):
        self.products[product.name] = product
        self.prices[product.name] = price

    def get_unit_price(self, product):
        return self.prices[product.name]
